### ISSUE TITLE
rule: `prefer-type-annotations`

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#203]
## Description
Prefer using type annotations for contextual types over `@as` where possible. Idea provided by [Inspector Boat in Zig's discord server](https://discord.com/channels/605571803288698900/719644313348341760/1329323459091894345).

## Examples

Examples of **incorrect** code for this rule
```zig
const foo = @as(T, @castBuiltin(bar)); 
```

Examples of **correct** code for this rule
```zig
const foo: T = @castBuiltin(bar)
```
